backend/src/services/user/user.service.ts

```diff
@@ -89,29 +92,44 @@ class User {
     };
   }
 
+  async getUserPermissions(
+    id: number,
+  ): Promise<PermissionsGetAllItemResponseDto[]> {
+    return await this.#userRepository.getUserPermissions(id);
+  }
+
   async getById(id: string): Promise<UsersByIdResponseDto | null> {
     const user = await this.#userRepository.getById(id);
 
     if (!user) {
       return null;
     }
+    const permissions = await this.#userRepository.getUserPermissions(user.id);
 
     return {
       id: user.id,
       email: user.email,
       fullName: user.fullName,
       createdAt: user.createdAt,
+      permissions: permissions,
     };
   }
 
   async getByIds(ids: number[]): Promise<UsersByIdResponseDto[]> {
     const users = await this.#userRepository.getByIds(ids);
 
-    return users.map((user) => ({
+    const usersPermissions = users.map(
+      async (user) => await this.#userRepository.getUserPermissions(user.id),
+    );
```

Comment:
If you want to add users permissions to the users by ids results, you can enhance `getByIds` query with the same join as in `getUserPermissions`(note that then you'll need to point correctly what columns that query should return)